llama_index/callbacks/open_inference_callback.py

```diff
@@ -0,0 +1,315 @@
+"""
+Callback handler for retrieval-augmented generation data in OpenInference
+format.
+"""
+
+import hashlib
+import importlib
+import os
+import sys
+from collections import OrderedDict
+from dataclasses import dataclass, field, fields
+from datetime import datetime
+from types import ModuleType
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Generic,
+    List,
+    Optional,
+    TypeAlias,
+    TypeVar,
+)
+
+from llama_index.callbacks.base import BaseCallbackHandler
+from llama_index.callbacks.schema import CBEventType, EventPayload
+
+if TYPE_CHECKING:
+    from pandas import DataFrame
```

Comment:
not needing this entirely could be nice if you are not accumulating data in pandas

Reply:
refactored in [88381c8](https://github.com/jerryjliu/llama_index/pull/6998/commits/88381c87e17cfcf12ad7db2766c3881ccdcb1a6d)